### PR TITLE
[pt2] add metas for `median` ops

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #106273
* __->__ #106272

### CODE REVIEW

**Comment by nkaretnikov** [comment]
From the list in https://github.com/pytorch/pytorch/issues/105105

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -3042,17 +3042,24 @@ def meta_nansum(input, dims=None, keepdim=False, *, dtype=None):
     return input.new_empty(output_shape, dtype=output_dtype)
 
 
-@register_meta(aten.nanmedian.default)
-def meta_nanmedian(input):
+@register_meta([aten.median.default, aten.nanmedian.default])
+def meta_median(input):
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:3046]: median is the same as nanmedian without NaN handling

File: torch/_meta_registrations.py [hunk 2]
```diff
@@ -3042,17 +3042,26 @@ def meta_nansum(input, dims=None, keepdim=False, *, dtype=None):
...
+        aten.median.dim_values,
+        aten.nanmedian.dim,
+        aten.nanmedian.dim_values,
+    ]
+)
 @out_wrapper("values", "indices")
-def meta_nanmedian_dim(input, dim=-1, keepdim=False):
+def meta_median_dim(input, dim=-1, keepdim=False):
+    if device_hint(input) == "cuda":
+        utils.alert_not_deterministic("median CUDA with indices output")
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:3064]: Repro (this alert is only on CUDA):
```
PYTORCH_TEST_WITH_INDUCTOR=1  python -bb test/test_torch.py -v --use-pytest --import-slow-tests --import-disabled-tests -k test_nondeterministic_alert_median -v --capture=no
```

**Comment by nkaretnikov** [comment]
The XLA failure is real and will require updating tests on the XLA side.

**Comment by nkaretnikov** [comment]
Updated XLA pin to point to companion PR.

**Review by ezyang — APPROVED** [review]

**Comment by nkaretnikov** [comment]
XLA people told me the issue on the XLA side needs to be fixed in a different way. Converting to a draft, so it's not merged by accident.

**Pull request closed without merging**